purchase_stock_picking_return_invoicing/models/purchase_order.py

```diff
@@ -127,15 +127,35 @@ def _compute_qty_refunded(self):
         'move_ids.returned_move_ids.state',
     )
     def _compute_qty_returned(self):
+        """Made through read_group for not impacting in performance."""
+        ProductUom = self.env['product.uom']
+        StockMove = self.env['stock.move']
+        groups = StockMove.read_group(
+            [('origin_returned_move_id.purchase_line_id', 'in', self.ids),
+             ('state', '=', 'done'),
+             ('location_id.usage', '!=', 'supplier')],
+            ['origin_returned_move_id', 'product_uom_qty', 'product_uom'],
+            ['origin_returned_move_id', 'product_uom'], lazy=False,
+        )
+        p = self._prefetch
+        # load all moves records at once on first access
+        origin_move_ids = set(g['origin_returned_move_id'][0] for g in groups)
```

Comment:
Can you check this option?

        p = self._prefetch
        group_dict = defaultdict(list)
        uom_ids = set()
        for g in groups:
            group_dict[g['origin_returned_move_id'][0]].append(
                g['product_uom'][0], g['product_uom_qty']
            )
            uom_ids.add(g['product_uom'][0])
        ProductUom.browse(list(uom_ids)).with_prefetch(p)
        for line in self:
            qty = 0.0
            for gl in group_dict[line.id]:
                if gl[0] == line.product_uom.id:
                    qty += gl[1]
                else:
                    uom = ProductUom.browse(gl[0]).with_prefetch(p)
                    qty += line.product_uom
                    x = uom._compute_quantity(gl[1], line.product_uom)
            line.qty_returned = qty